lib/azure/armrest/configuration.rb

```diff
@@ -124,7 +124,7 @@ def initialize(args)
         # Delay this to avoid a double call
         @environment = options.delete(:environment)
 
-        if @environment.to_s.casecmp(Azure::Armrest::USGOV_ENVIRONMENT) == 0
+        if @environment.to_s.downcase == Azure::Armrest::USGOV_ENVIRONMENT
```

Comment:
We don't need any special handling for `@environment`, just let it naturally fall through the setter for `environment`. So you can delete line 109, 110, 125, and block 127.